samples/suit/flash_companion/README.rst

```diff
@@ -0,0 +1,115 @@
+.. _suit_flash_companion:
+
+SUIT flash companion
+####################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The flash companion application allows the secure domain firmware to access the external memory during the :ref:`Software Updates for Internet of Things (SUIT) <ug_nrf54h20_suit_dfu>` firmware upgrade.
+
+Requirements
+************
+
+The application supports the following development kit:
+
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :rows: nrf54h20dk/nrf54h20/cpuapp
+
+Overview
+********
+
+The flash companion implements a device driver for the external flash device and provides an IPC service to the secure domain.
+The secure domain firmware can request from the flash companion to read, erase, or write data to the external memory.
+
+The application is meant to be booted by the secure domain while performing the firmware update process using the :ref:`SUIT <ug_nrf54h20_suit_dfu>` firmware upgrade.
+
+The flash companion firmware is not a stand-alone application, it is intended to be used with the ``nrf54h_suit_sample`` to complete a firmware transfer with external flash.
+The application must be built as a child image of a different sample as it inherits the devicetree configuration from the parent build.
+
+Configuration
+*************
+
+|config|
+
+Setup
+=====
+
+You can build the application using Sysbuild by enabling the :kconfig:option:`SB_CONFIG_SUIT_BUILD_FLASH_COMPANION` Kconfig option.
+The memory partition from which the firmware will run can be configured by providing a device tree overlay through Sysbuild.
+You should create a dedicated partition in non-volatile memory and override the ``zephyr,code-partition``.
+The memory partition must not be used by anything else.
```

Comment:
Can it be more specific rather than "_anything else_"?

Reply:
Rephrased to `must not be used by any other firmware image`